src/ert/storage/local_ensemble.py

```diff
@@ -574,13 +662,122 @@ def load_parameters(
+            drop_reals_dim = True
+            selected_realizations = realizations
+        elif type(realizations) is np.ndarray:
+            selected_realizations = realizations.tolist()
+        elif isinstance(realizations, tuple):
+            selected_realizations = list(realizations)
+        elif isinstance(realizations, list):
+            selected_realizations = realizations
+        else:
+            raise ValueError(f"Invalid type for realizations: {type(realizations)}")
+
+        try:
+            ds = self.open_unified_parameter_dataset(group)
+            if selected_realizations is None:
+                return ds
+
+            return ds.sel(realizations=selected_realizations, drop=drop_reals_dim)
+
+        except (ValueError, KeyError, FileNotFoundError):
+            # Fallback to check for real folder
+            try:
+                if selected_realizations is None:
+                    return xr.combine_nested(
+                        [
+                            xr.open_dataset(p)
+                            for p in self._path.glob(f"realization-*/{group}.nc")
+                        ],
+                        concat_dim="realizations",
+                    )
+                elif type(selected_realizations) is int:
+                    return xr.open_dataset(
+                        self._path
+                        / f"realization-{selected_realizations}"
+                        / f"{group}.nc"
+                    )
+                else:
+                    assert type(selected_realizations) is list
+                    return xr.combine_nested(
+                        [
+                            xr.open_dataset(
+                                self._path / f"realization-{real}" / f"{group}.nc"
+                            )
+                            for real in selected_realizations
+                        ],
+                        concat_dim="realizations",
+                    )
+            except FileNotFoundError as e:
+                raise KeyError(
+                    f"No dataset '{group}' in storage for "
+                    f"realization {selected_realizations}"
+                ) from e
+
+    def _find_unified_dataset_for_response(self, key: str) -> str:
+        all_gen_data_keys = {
+            k
+            for k, c in self.experiment.response_configuration.items()
+            if type(c) is GenDataConfig
+        }
+
+        if key == "gen_data" or key in all_gen_data_keys:
+            return "gen_data"
+
+        if key == "summary" or key in self.get_summary_keyset():
+            return "summary"
+
+        if key not in self.experiment.response_configuration:
+            raise ValueError(f"{key} is not a response")
+
+        return key
+
+    def open_unified_response_dataset(self, key: str) -> xr.Dataset:
+        dataset_key = self._find_unified_dataset_for_response(key)
+        nc_path = self._path / f"{dataset_key}.nc"
+
+        ds = None
```

Comment:
Is this needed? You are checking with `_find_unified_dataset_for_response`

Reply:
(assuming you are referring to the check if the ds exists) 
At the moment I think so yep, because _find_unified_dataset_for_response finds the unified dataset "key" corresponding to the requested key (which may be for example "gen_data" or a specific key within gen data), but it doesn't check if it actually exists.